core-bundle/src/Controller/FrontendModule/RootPageDependentModuleController.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller\FrontendModule;
+
+use Contao\Controller;
+use Contao\CoreBundle\Controller\AbstractFragmentController;
+use Contao\ModuleModel;
+use Contao\PageModel;
+use Contao\StringUtil;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class RootPageDependentModuleController extends AbstractFragmentController
+{
+    public function __invoke(Request $request, ModuleModel $model, string $section, array $classes = null): Response
+    {
+        /** @var PageModel $pageModel */
+        $pageModel = $this->getPageModel();
+
+        $controller = $this->container->get('contao.framework')->getAdapter(Controller::class);
+
+        $modules = StringUtil::deserialize($model->rootPageDependentModules);
+        $content = '';
+
+        if (\is_array($modules) && \array_key_exists($pageModel->rootId, $modules)) {
+            $content = $controller->getFrontendModule($modules[$pageModel->rootId]);
+        }
+
+        $response = new Response($content);
+
+        $this->markResponseForInternalCaching($response);
```

Comment:
this makes no sense since the controller does not include caching information anyway.
Also the rendered module might come from a fragment response that already affected caching (not sure if that works, but that would be a separate issue).

Reply:
cc @Toflar

Reply:
It does make sense but the tags are missing. When you edit this FE module, the pages where this module is placed should be invalidated. So the problem here is not the `markResponseForInternalCaching()`, that's correct. The problem is the missing `tagResponse()` call.

Reply:
Add tag in 971a70a7f.

Reply:
`markResponseForInternalCaching` tells our fragment handler to merge `Cache-Control` headers, if the response has any. Calling it is useful if you do add cache headers, or if one does not know if the parent method does (that's why the twig render method calls it). It is not useful if you do not add any caching info to the response (as this module does).

Reply:
Hmm true, you would actually kill caching now because the default is `private, no-store` 🙈  So that has to go again. But I think I'd leave the changes in abstract anyway because there surely are other use cases where a controller would want to use that method.

Reply:
So remove the tag again?

Reply:
Nooo, the tag is correct, the cache call is not. I'll suggest.

Reply:
Fixed in 4d18783.